types/src/front/lib/assistant.ts

```diff
@@ -1,29 +1,85 @@
+import { ExtractSpecificKeys } from "../../shared/typescipt_utils";
+
 /**
- * Supported models
+ * PROVIDER IDS
  */
 
-import { ExtractSpecificKeys } from "../../shared/typescipt_utils";
+export const MODEL_PROVIDER_IDS = [
+  "openai",
+  "anthropic",
+  "mistral",
+  "google_ai_studio",
+] as const;
+export type ModelProviderIdType = (typeof MODEL_PROVIDER_IDS)[number];
+
+export const isModelProviderId = (
+  providerId: unknown
```

Comment:
Should it be `string` at least? Otherwise, you need to add `typeof providerId === "string"`.